utils/report.c

```diff
@@ -752,6 +823,36 @@ FIELD_TIME(REPORT_F_SELF_TIME_MIN, self-min, self.min, self_min, "Self min", 10)
 FIELD_TIME(REPORT_F_SELF_TIME_MAX, self-max, self.max, self_max, "Self max", 10);
 FIELD_CALL(REPORT_F_CALL, call, call, call, "Calls", 10);
 
+FIELD_TIME_DIFF(REPORT_F_TOTAL_TIME, total, total.sum, total_diff, "Total time", 11);
+FIELD_TIME_DIFF(REPORT_F_TOTAL_TIME_AVG, total-avg, total.avg, total_avg_diff, "Total avg", 11);
+FIELD_TIME_DIFF(REPORT_F_TOTAL_TIME_MIN, total-min, total.min, total_min_diff, "Total min", 11);
+FIELD_TIME_DIFF(REPORT_F_TOTAL_TIME_MAX, total-max, total.max, total_max_diff, "Total max", 11);
+FIELD_TIME_DIFF(REPORT_F_SELF_TIME, self, self.sum, self_diff, "Self time", 11);
+FIELD_TIME_DIFF(REPORT_F_SELF_TIME_AVG, self-avg, self.avg, self_avg_diff, "Self avg", 11);
+FIELD_TIME_DIFF(REPORT_F_SELF_TIME_MIN, self-min, self.min, self_min_diff, "Self min", 11);
+FIELD_TIME_DIFF(REPORT_F_SELF_TIME_MAX, self-max, self.max, self_max_diff, "Self max", 11);
```

Comment:
You can move the common suffix like `_diff` into the macro definition (`FIELD_TIME_DIFF`) to reduce duplication..  

```c
#FIELD_TIME_DIFF(...) \
static void print ## _func ## _diff(...) \
FIELD_STRUCT(... , _func ## _diff, ...)
```

Likewise, if length is all same, let's move it to the macro too.